dev-requirements.txt

```diff
@@ -21,9 +21,13 @@ async-timeout==4.0.2  # Used for timeouts in some test cases.
 # DOCUMENTATION #
 #################
 
-pdoc==11.0.0
-sphobjinv==2.2.2
-minify-html==0.8
+sphinx==5.0.1
+sphinx-autoapi==1.8.4
```

Comment:
Attempt to get rid of this dependency

Reply:
Better do not delete `sphinx-autoapi`. Build-in auto API tool requires write reference to every module by hand. And then it's importing everything for documentation, while `sphinx-autoapi` parsing source code and automaticly find all source files, as expected from such tool.

You can generate *stubs* for build-in API tool on every run, so no need to reference in docs every module. **BUT** this will **import** the code plus it's too slow and often breaks.

But `sphinx-autoapi` also have some dissadvantages, like unsuporting redefing (althought mypy also doesn't support it).

Reply:
I left this purely as a note to remind me what to finish. I was making an alternative to it that works better in some parts of the code that this currently breaks, mainly enums (off the top of my head), but I still got to play around with it a bit